src/test/java/DeckApiServisTest.java

```diff
@@ -0,0 +1,538 @@
+
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.example.Card;
+import org.example.DeckApiServis;
+import org.example.DeckToken;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.io.IOException;
+import java.net.http.HttpResponse;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.example.Suit.HEARTS;
+import static org.example.Suit.SPADES;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class DeckApiServisTest {
```

Comment:
Please seperate mapping functionality from API service, then I will review tests in depth

Reply:
ok , przemodeluje klasy